nntrainer/compiler/multiout_realizer.cpp

```diff
@@ -9,16 +9,99 @@
  * @author Jihoon Lee <jhoon.it.lee@samsung.com>
  * @bug No known bugs except for NYI items
  */
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+
+#include <common_properties.h>
 #include <compiler_fwd.h>
+#include <layer_node.h>
 #include <multiout_realizer.h>
+#include <remap_realizer.h>
 
 namespace nntrainer {
 MultioutRealizer::~MultioutRealizer() {}
 
 GraphRepresentation
 MultioutRealizer::realize(const GraphRepresentation &reference) {
+  GraphRepresentation processed(reference.begin(), reference.end());
+
+  std::unordered_multiset<std::string> freq_map;
+  std::unordered_set<std::string> node_names;
+  std::unordered_set<std::string> connection_names;
+
+  /// 1. build frequency map and connection names
+  for (auto &node : reference) {
+    NNTR_THROW_IF(node_names.count(node->getName()), std::invalid_argument)
+      << "node name clashes: " << node->getName();
+    node_names.emplace(node->getName());
+
+    for (unsigned int i = 0, num_nodes = node->getNumInputConnections();
+         i < num_nodes; ++i) {
+      props::InputConnection c(props::Connection(
+        node->getInputConnectionName(i), node->getInputConnectionIndex(i)));
+      auto uniq_name = to_string(c);
+      connection_names.emplace(uniq_name);
+      freq_map.emplace(uniq_name);
+    }
+  }
+
+  /// 2. for each connection names, if a connection is referenced multiple
+  /// times, create multioutput node and remap to multi output node index
+  std::unordered_multimap<std::string /**< original id */,
```

Comment:
multimap allows multiple entries of equivalent keys, but L70 ensures that the key is unique. So, this can be unordered_map?

Reply:
there will be 1:n mapping between ${id(layer name)}:${multiout_name}

consider below ini

```
[lstm]
type=lstmcell

[a]
input_layers=lstm(0)

[b]
input_layers=lstm(0)

[c]
input_layers=lstm(1)

[c]
input_layers=lstm(1)
```

key lstm will have lstm/generated_out_0, and lstm/generated_out_1

Reply:
This is fine, I thought the the struct is `multimap` of `string : [node]` than `string : node`.
Thanks for the example.